### PR TITLE
Remove dependence on unsupported pip API

### PR DESCRIPTION
**Pull request by theo-o**
In pip>=10.0.0, the pip team got rid of access to pip.download and pip.req  ([https://github.com/pypa/pip/issues/4696](https://github.com/pypa/pip/issues/4696)).  Currently, builds are failing due to this issue.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
 
+def parse_requirements(filename):
+    lineiter = [line.strip() for line in open(filename)]
+    return [line for line in lineiter if line and not line.startswith("#")]
+
+
 def get_requirements():
     rtd_file = open('docs/rtd-requirements.txt', 'w')
-    for dep in parse_requirements('requirements.txt', session=PipSession()):
-        # FIXME: there should really be a better way to handle this...
```
Comment (pefoley2) [inline setup.py]: Glad to see someone finally dealt with my god-awful hack :)

File: .gitignore [hunk 2]
```diff
@@ -30,3 +30,5 @@ fixtures/*.sql
 fixtures/README.txt
 
 *fixtures*.tar*
+.virtualenvs
+.sass-cache
```
Comment (pefoley2) [inline .gitignore]: These seem unrelated...

File: intranet/apps/signage/migrations/0015_auto_20180420_1825.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (pefoley2) [inline intranet/apps/signage/migrations/0015_auto_20180420_1825.py]: As do these

File: setup.py [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
-from pip.req import parse_requirements
-
 from setuptools import find_packages, setup
 
 with open('README.rst', 'r') as f:
     long_description = f.read()
 
 
+def parse_requirements(filename):
+    lineiter = [line.strip() for line in open(filename)]
```
Comment (pefoley2) [inline setup.py]: Use a with block to avoid leaking the fd?

File: setup.py [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
+    return [line for line in lineiter if line and not line.startswith("#")]
+
+
 def get_requirements():
     rtd_file = open('docs/rtd-requirements.txt', 'w')
```
Comment (pefoley2) [inline setup.py]: Why not make this a with block as well to avoid having to call close explicitly?

File: intranet/apps/signage/migrations/0015_auto_20180420_1825.py (hunk 3 above)
Comment (theo-o) [reply intranet/apps/signage/migrations/0015_auto_20180420_1825.py]: The build fails without it.

File: .gitignore (hunk 2 above)
Comment (theo-o) [reply .gitignore]: Forgot to drop this.

File: setup.py (hunk 5 above)
Comment (theo-o) [reply setup.py]: Good idea.  Looking at the code again, a compound with statement does make more sense.

File: intranet/apps/signage/migrations/0015_auto_20180420_1825.py (hunk 3 above)
Comment (pefoley2) [reply intranet/apps/signage/migrations/0015_auto_20180420_1825.py]: Sure, but it should really be a separate PR as it's completely unrelated.

**Comment by theo-o** [comment]
Retargeted and rebased
